src/account-info/AccountDetailsScreen.js

```diff
@@ -30,7 +30,9 @@ type SelectorProps = $ReadOnly<{|
 |}>;
 
 type Props = $ReadOnly<{|
-  navigation: NavigationScreenProp<{ params: {| userId: number |} }>,
+  // Since we've put this screen in a stack-nav route config, it gets the
+  // `navigation` prop for free.
+  navigation: NavigationStackProp<{| ...NavigationStateRoute, params: {| userId: number |} |}>,
```

Comment:
The main example in the TypeScript doc is the following:
    
    ```
    type Props = {
      navigation: NavigationStackProp<{ userId: string }>;
    };
    ```
    
    We find `NavigationStackProp` is a good thing to use, but
    `{ userId: string }` is bad for a couple of reasons:
    
    - It's off by a level of nesting; surely they mean to describe
      `navigation.state.params`. To do that, the type needs to look
      something like `{ params: { userId: string } }`.
    
    - It leaves out all the other things on `navigation.state` including
      `key` and `routeName`, which might be nice to have someday.
      Experimentally, these are conveniently filled in by saying
      `NavigationStackProp<NavigationStateRoute>` [3].

Yeah, that's odd!

I wondered if this might be a difference between the TS definitions and the Flow libdef, in what they have `NavigationStackProp` take as a type parameter. But AFAICT it's equally wrong there -- the first type parameter is called `State` and is used for the `state` property, not `state.params`. (Though there is a *second* type parameter for the latter.)

Reply:
Right, I did see some wrong things in the TypeScript (and noted them briefly in the commit message) but didn't record any specifics like this.